lib/src/source_visitor.dart

```diff
@@ -2599,6 +2623,53 @@ class SourceVisitor extends ThrowingAstVisitor {
     visit(node.operand);
   }
 
+  @override
+  void visitRepresentationConstructorName(RepresentationConstructorName node) {
+    token(node.period);
+    token(node.name);
+  }
+
+  @override
+  void visitRepresentationDeclaration(RepresentationDeclaration node) {
+    // Consider having `node.asFormalParameterList()`, and format based on that.
+
+    visit(node.constructorName);
+
+    token(node.leftParenthesis);
+
+    final rule = PositionalRule(null, argumentCount: 1);
+
+    builder.startRule(rule);
+    rule.beforeArgument(zeroSplit());
+
+    // Make sure record and function type parameter lists are indented.
+    builder.startBlockArgumentNesting();
+    builder.startSpan();
+
+    // Inline `visitFormalParameterList`, remove all that doesn't apply
```

Comment:
[nit] This and the following comments are useful during the review but will lose relevance because they are so temporally specific. I think they can be removed from the source and added to the commit message.